databricks/koalas/frame.py

```diff
@@ -7987,7 +7987,12 @@ def _reindex_columns(self, columns, fill_value):
                 scols.append(F.lit(fill_value).alias(name_like_string(label)))
             labels.append(label)
 
-        return DataFrame(self._internal.with_new_columns(scols, column_labels=labels))
+        df = DataFrame(self._internal.with_new_columns(scols, column_labels=labels))
+
+        if isinstance(columns, pd.Index):
+            return df.rename_axis(columns=columns.names)
```

Comment:
How about `df.rename_axis(columns=columns.names, inplace=True)` and returning `df` instead? I believe this would save us from creating a new DataFrame.

Reply:
We should rather specify `columns.names` when `self._internal.with_new_columns`.

Reply:
Yes indeed !